reco_utils/dataset/python_splitters.py

```diff
@@ -162,17 +183,26 @@ def python_stratified_split(
             col_item=col_item,
         )
 
-    num_of_splits = len(ratio)
-    splits = [pd.DataFrame({})] * num_of_splits
+    # Split by each group and aggregate splits together.
+    splits = []
     df_grouped = data.groupby(split_by_column)
     for name, group in df_grouped:
         group_splits = split_pandas_data_with_ratios(
             df_grouped.get_group(name), ratio, resample=True, seed=seed
         )
-        for x in range(num_of_splits):
-            splits[x] = pd.concat([splits[x], group_splits[x]])
 
-    return splits
+        # Concatenate the list of split dataframes.
+        concat_group_splits = pd.concat(group_splits)
+
+        splits.append(concat_group_splits)
+
+    # Concatenate splits for all the groups together.
+    splits_all = pd.concat(splits)
+
+    # Take split by split_index
+    splits_list = [splits_all[splits_all['split_index'] == x].drop('split_index', axis=1) for x in range(len(ratio))]
+
```

Comment:
Our stratified and chrono splitters share many similar codes. Does it make sense to define a private function, e.g. `_prepare_stratification()` and share them from both splitters? Maybe a bit tricky in the main splitting algorithm part but at least the column validity check and data preparation part look the same.

Reply:
Good point, they look very similar. The main difference between the two methods is `sort_values(col_timestamp)`
Probably the chrono split could be reduced to sorting the timestamps and then calling the stratified splitter. We could open a new issue for this.

Reply:
I created an issue and added you https://github.com/Microsoft/Recommenders/issues/537